parser-typechecker/unison/Main.hs

```diff
@@ -309,32 +200,34 @@ runTranscripts' codebaseFormat mcodepath transcriptDir args = do
       pure False
 
 runTranscripts
-  :: CodebaseFormat
-  -> Bool
-  -> Bool
+  :: UsageRenderer
+  -> CodebaseFormat
+  -> ShouldForkCodebase
+  -> ShouldSaveCodebase
   -> Maybe FilePath
   -> [String]
   -> IO ()
-runTranscripts cbFormat inFork keepTemp mcodepath args = do
+runTranscripts renderUsageInfo cbFormat shouldFork shouldSaveTempCodebase mcodepath args = do
   progName <- getProgName
-  transcriptDir <- prepareTranscriptDir cbFormat inFork mcodepath
+  transcriptDir <- prepareTranscriptDir cbFormat shouldFork mcodepath
   completed <-
     runTranscripts' cbFormat (Just transcriptDir) transcriptDir args
-  when completed $ do
-    unless keepTemp $ removeDirectoryRecursive transcriptDir
-    when keepTemp $ PT.putPrettyLn $
-        P.callout "🌸" (
-          P.lines [
-            "I've finished running the transcript(s) in this codebase:", "",
-            P.indentN 2 (P.string transcriptDir), "",
-            P.wrap $ "You can run"
-                  <> P.backticked (P.string progName <> " -codebase " <> P.string transcriptDir)
-                  <> "to do more work with it."])
-
-  unless completed $ do
-      unless keepTemp $ removeDirectoryRecursive transcriptDir
-      PT.putPrettyLn (usage progName)
-      Exit.exitWith (Exit.ExitFailure 1)
+  case shouldSaveTempCodebase of
```

Comment:
Note that there's a small re-arrangement here which _shouldn't_ affect behaviour, but is worth looking closer at 😄